internal/rbd/rbd_attach.go

```diff
@@ -193,35 +204,46 @@ func waitForPath(ctx context.Context, pool, namespace, image string, maxRetries
 	return "", false
 }
 
-// Check if rbd-nbd tools are installed.
-func checkRbdNbdTools() bool {
+// getRbdNbdToolsVersion returns rbd-nbd tools version.
+func getRbdNbdToolsVersion() string {
 	// check if the module is loaded or compiled in
 	_, err := os.Stat(fmt.Sprintf("/sys/module/%s", moduleNbd))
 	if os.IsNotExist(err) {
 		// try to load the module
 		_, _, err = util.ExecCommand(context.TODO(), "modprobe", moduleNbd)
 		if err != nil {
-			log.ExtendedLogMsg("rbd-nbd: nbd modprobe failed with error %v", err)
+			log.WarningLogMsg("rbd-nbd: nbd modprobe failed with error %v", err)
 
-			return false
+			return ""
 		}
 	}
-	if _, _, err := util.ExecCommand(context.TODO(), rbdTonbd, "--version"); err != nil {
-		log.ExtendedLogMsg("rbd-nbd: running rbd-nbd --version failed with error %v", err)
+	stdout, stderr, err := util.ExecCommand(context.TODO(), rbdTonbd, "--version")
+	if err != nil || stderr != "" {
+		log.WarningLogMsg("rbd-nbd: running rbd-nbd --version failed with error:%v, stderr:%s", err, stderr)
+
+		return ""
+	}
+	ver := strings.Split(stdout, " ")
+	// ex: ceph version 15.2.14 cd3bb7e87a2f62c1b862ff3fd8b1eec13391a5be) octopus (stable)
+	rbdNbdVer := ver[2]
```

Comment:
this can cause panic if the ver length is not 3? need a  length check?

Reply:
oops! right, I thought I added one based on your previous comment, will fix it.